src/components/input-number/input-number.tsx

```diff
@@ -0,0 +1,884 @@
+   */
+  @Prop({ reflect: true }) clearable = false;
+
+  /**
+   * When true, interaction is prevented and the component is displayed with lower opacity.
+   *
+   * @mdn [disabled](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled)
+   */
+  @Prop({ reflect: true }) disabled = false;
+
+  @Watch("disabled")
+  disabledWatcher(): void {
+    this.setDisabledAction();
+  }
+
+  /**
+   * When true, number values are displayed with the locale's group separator.
+   */
+  @Prop() groupSeparator = false;
+
+  /**
+   * When true, the component will not be visible.
+   *
+   * @mdn [hidden](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden)
+   */
+  @Prop() hidden = false;
+
+  /**
+   * When true, shows a default recommended icon. Alternatively, pass a Calcite UI Icon name to display a specific icon.
+   */
+  @Prop({ reflect: true }) icon: string | boolean;
+
+  /**
+   * A text label that will appear on the clear button for screen readers.
+   */
+  @Prop() intlClear?: string;
+
+  /**
+   * Accessible name that will appear while loading.
+   *
+   * @default "Loading"
+   */
+  @Prop() intlLoading?: string = COMMON_TEXT.loading;
+
+  /** When true, the icon is flipped in RTL. */
+  @Prop({ reflect: true }) iconFlipRtl = false;
+
+  /** Accessible name for the component's button or hyperlink. */
+  @Prop() label?: string;
+
+  /** When true, the component is in the loading state and `calcite-progress` is displayed. */
+  @Prop({ reflect: true }) loading = false;
+
+  /** Specifies the BCP 47 language tag for the desired language and country format. */
+  @Prop() locale: string = document.documentElement.lang || "en";
+
+  /**
+   * Specifies the Unicode numeral system used by the component for localization.
+   *
+   * @mdn [numberingSystem](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/numberingSystem)
+   */
+  @Prop() numberingSystem?: string;
+
+  /**
+   * Toggles locale formatting for numbers.
+   *
+   * @internal
+   */
+  @Prop() localeFormat = false;
+
+  /**
+   * Specifies the maximum value.
+   *
+   * @mdn [max](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#max)
+   */
+  @Prop({ reflect: true }) max?: number;
+
+  /** watcher to update number-to-string for max */
+  @Watch("max")
+  maxWatcher(): void {
+    this.maxString = this.max?.toString() || null;
+  }
+
+  /**
+   * Specifies the minimum value.
+   *
+   * @mdn [min](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#min)
+   */
+  @Prop({ reflect: true }) min?: number;
+
+  /** watcher to update number-to-string for min */
+  @Watch("min")
+  minWatcher(): void {
+    this.minString = this.min?.toString() || null;
+  }
+
+  /**
+   * Specifies the maximum length of text for the component's value.
+   *
+   * @deprecated use maxLength instead
+   */
+  @Prop({ reflect: true }) maxlength?: number;
```

Comment:
Since this is a brand new component and `calcite-input` will remain intact for a while, do we need to include this deprecated property?